src/main/java/com/fasterxml/jackson/databind/util/NativeImageUtil.java

```diff
@@ -9,31 +9,39 @@
  * @since 2.14
  */
 public class NativeImageUtil {
-    private static final boolean RUNNING_IN_SVM;
-
-    static {
-        RUNNING_IN_SVM = System.getProperty("org.graalvm.nativeimage.imagecode") != null;
-    }
+    
+    private static final boolean RUNNING_IN_SVM = isInNativeImage();
 
     private NativeImageUtil() {
     }
 
     /**
-     * Check whether we're running in substratevm native image runtime mode.
-     * This check cannot be a constant, because
+     * Check whether we're running in SubstrateVM native image and also in "runtime" mode.
+     * The "runtime" check cannot be a constant, because
      * the static initializer may run early during build time
      *<p>
-     * NOTE: {@code public} since 2.16 (before that, {@code private}).
+     * As optimization, {@link #RUNNING_IN_SVM} is used to short-circuit on normal JVMs.
      */
-    public static boolean isRunningInNativeImage() {
+    private static boolean isInNativeImageAndIsAtRuntime() {
         return RUNNING_IN_SVM && "runtime".equals(System.getProperty("org.graalvm.nativeimage.imagecode"));
     }
 
+    /**
+     * Checks whether we're running in SubstrateVM native image <b>only by the presence</b> of 
+     * {@code "org.graalvm.nativeimage.imagecode"} system property, regardless of its value (buildtime or runtime).
+     * We are irrespective of the build or runtime phase, because native-image can initialize static initializers at build time.
+     *<p>
+     * @since 2.16
+     */
+    public static boolean isInNativeImage() {
+        return System.getProperty("org.graalvm.nativeimage.imagecode") != null;
```

Comment:
this check *can* be inlined into RUNNING_IN_SVM, because it doesnt care whether the value is buildtime or runtime.

Reply:
Ok so this should just return `RUNNING_IN_SVM`, right?

Reply:
yep